pkg/traces/config_test.go

```diff
@@ -1732,10 +1732,46 @@ func sortPipelines(cfg *config.Config) {
 		exp         = tracePipeline.Exporters
 		recv        = tracePipeline.Receivers
 		ext         = cfg.Service.Extensions
+		pipelines   = cfg.Pipelines
 		serviceRecv = cfg.Service.Pipelines[config.NewComponentID(config.TracesDataType)].Receivers
 	)
-	sort.Slice(exp, func(i, j int) bool { return exp[i].String() > exp[j].String() })
-	sort.Slice(recv, func(i, j int) bool { return recv[i].String() > recv[j].String() })
-	sort.Slice(ext, func(i, j int) bool { return ext[i].String() > ext[j].String() })
-	sort.Slice(serviceRecv, func(i, j int) bool { return serviceRecv[i].String() > serviceRecv[j].String() })
+	sort.SliceStable(exp, func(i, j int) bool { return exp[i].String() > exp[j].String() })
+	sort.SliceStable(recv, func(i, j int) bool { return recv[i].String() > recv[j].String() })
+	sort.SliceStable(ext, func(i, j int) bool { return ext[i].String() > ext[j].String() })
+	sort.SliceStable(serviceRecv, func(i, j int) bool { return serviceRecv[i].String() > serviceRecv[j].String() })
+	for _, pipeline := range pipelines {
+		sort.SliceStable(pipeline.Exporters, func(i, j int) bool { return pipeline.Exporters[i].String() > pipeline.Exporters[j].String() })
+		sort.SliceStable(pipeline.Receivers, func(i, j int) bool { return pipeline.Receivers[i].String() > pipeline.Receivers[j].String() })
+		sort.SliceStable(pipeline.Processors, func(i, j int) bool { return pipeline.Processors[i].String() > pipeline.Processors[j].String() })
+	}
+}
+
+func assertConfigEqual(t *testing.T, cfg1, cfg2 *config.Config) {
```

Comment:
Thanks for fixing the tests that errored with the upgrade of prometheus/common!

Sorry that you had to spend time on this as well. I tried to see whether this fixes the flakiness once and for all; it seems it's a lot better but still doesn't 100% pass. 
```
$ go test -run=TestOTelConfig -count=100 | grep "FAIL" | wc -l
      38
```

My 2c is to leave this another PR, keep track of it on #1654 and only focus on the SSL exporter. WDYT? I'll get the word out to people more experienced with the Tracing side of things; I have a hunch that there's a deeper issue with how we use pointers as map keys in the OTEL configuration but they can let us know.

Reply:
hey it's been a fun problem to (try to) solve 😎 

That sounds good to me @tpaschalis, should I keep my changes thus far or should I revert my changes to config_test.go?

Reply:
IMHO let's revert changes to `pkg/traces/config_test.go` and focus on the exporter itself.